common/djangoapps/third_party_auth/signals/handlers.py

```diff
@@ -0,0 +1,61 @@
+"""
+Signal handlers for third_party_auth app.
+"""
+
+from django.db.models.signals import post_save
+from django.dispatch import receiver
+from edx_django_utils.monitoring import set_custom_attribute
+
+
+from common.djangoapps.third_party_auth.models import SAMLConfiguration, SAMLProviderConfig
+
+
+from common.djangoapps.third_party_auth.toggles import ENABLE_SAML_CONFIG_SIGNAL_HANDLERS
+
+
+@receiver(post_save, sender=SAMLConfiguration)
+def update_saml_provider_configs_on_configuration_change(sender, instance, created, **kwargs):
+    """
+    NOTE: This behavior is controlled by the ENABLE_SAML_CONFIG_SIGNAL_HANDLERS toggle.
+
+    Signal handler to create a new SAMLProviderConfig when SAMLConfiguration is updated.
+
+    When a SAMLConfiguration is updated and a new version is created,
+    this handler generates a corresponding
+    SAMLProviderConfig that references the latest configuration version,
+    ensuring all providers remain aligned
+    with the most current settings.
```

Comment:
Keeping to 80 chars per line this should probably be max 3 lines. This odd formatting is just harder to read.